### PR TITLE
Add algorithm to get run information from ISIS journal files

### PR DESCRIPTION
**Pull request by gemmaguest**
**Description of work.**

This PR adds the algorithm `ISISJournalGetExperimentRuns`, which gets information about runs for a particular experiment from the ISIS journal files.

**Report to:** Max at ISIS

**To test:**

Must be tested on the ISIS network.

Try out the algorithm - see the usage information in its documentation. Check that the results in the output table make sense (you can check them e.g. against JournalViewer or ICat, although note that ICat is sometimes unreliable).

Refs #26897 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/inc/MantidDataHandling/ISISJournalGetExperimentRuns.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    return "ISISJournalGetExperimentRuns";
+  }
+  /// Summary of algorithms purpose
+  const std::string summary() const override {
+    return "Obtains information of the runs associated with a specific "
+           "investigation.";
+  }
+  /// Algorithm's version for identification overriding a virtual method
+  int version() const override { return 1; }
+  const std::vector<std::string> seeAlso() const override {
```
Comment (ConorMFinn) [inline Framework/DataHandling/inc/MantidDataHandling/ISISJournalGetExperimentRuns.h:40]: This seems to be the only declaration without a comment?

**Review by cailafinn — CHANGES_REQUESTED** [review]
Just a single code comment about a comment. And a couple of other notes I found while testing:
- The example for the cycle pop up is missing from the input dialog, appearing as: `the cycle name e` instead of `The cycle name e.g. 19_4`
- An invesigation ID where no runs are found simply returns an empty table workspace without warnings, is this the expected behaviour?

**Comment by gemmaguest** [comment]
> * The example for the cycle pop up is missing from the input dialog, appearing as: `the cycle name e` instead of `The cycle name e.g. 19_4`

Good spot! Apparently the full-stops were messing this up. I've removed them.

> * An invesigation ID where no runs are found simply returns an empty table workspace without warnings, is this the expected behaviour?

Yes, for now anyway - it might not be the best way but it's what `CatalogGetDataFiles` does and I was aiming for broadly similar behaviour.

**Review by cailafinn — APPROVED** [review]
Looks good to me now, nice work 👍

**Comment by gemmaguest** [comment]
@mantidproject/gatekeepers Could someone merge this because it's blocking another PR?

**Pull request merged**